src/run.ts

```diff
@@ -35,38 +40,21 @@ async function saveFile(filePath: string) {
     await vscode.workspace.textDocuments.find((doc) => doc.fileName === filePath)?.save();
 }
 
-export async function runInTerminal(text: string, filename: string) {
+export async function executeInTerminal(text: string, filename: string, testMode: "run" | "watch" = "run") {
     let terminalAlreadyExists = true;
     if (!terminal || terminal.exitStatus) {
         terminalAlreadyExists = false;
         terminal?.dispose();
         terminal = vscode.window.createTerminal(`vscode-vitest-runner`);
     }
 
-    const vitestArgs = buildVitestArgs({ caseName: text, casePath: filename });
-    const npxArgs = ['npx', ...vitestArgs];
-
-    if (terminalAlreadyExists) {
-        // CTRL-C to stop the previous run
-        terminal.sendText('\x03');
-    }
-
-    await saveFile(filename);
-
-    terminal.sendText(npxArgs.join(' '), true);
-    terminal.show();
-}
-
-export async function watchInTerminal(text: string, filename: string) {
-    let terminalAlreadyExists = true;
-    if (!terminal || terminal.exitStatus) {
-        terminalAlreadyExists = false;
-        terminal?.dispose();
-        terminal = vscode.window.createTerminal(`vscode-vitest-runner`);
-    }
+    const pretest = (config.get("pretest") as string[]);
+    const finalPretest = pretest.length > 0 ? [`${pretest.join("; ")};`] : []
```

Comment:
1. having `;` would not fail if pretest command fail please replace to `&&`
2. How other test runner extensions do this, this seems like can be security problem

Reply:
1. Done
2. The way test runner does is to simply change the jest command. So, regarding the security issue, I guess it's fine since theyt also allow you to that whatever and it's a pretty popular extension (1.000.000+ downloads) 🤷

![image](https://github.com/rluvaton/vitest-vs-code-plugin/assets/4213098/0361f293-3ec6-46eb-a175-adf1bf80ed2f)